### PR TITLE
docs(connector): update docs for connector

### PR DESCRIPTION
**Pull request by peiwangdb**
# Description

restructure and write new sections for connector

# How Has This Been Tested?

# Snapshots:

Include snapshots for easier review.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have already squashed the commits and make the commit message conform to the project standard.
- [ ] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/source/user_guide/connector/introduction.ipynb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Connector offers essential features to facilitate the process of collecting data, for example:\n",
+    "\n",
+    "* Concurrency: Collect data from websites, in parallel, in a fast way!\n",
+    "\n",
+    "* Pagination: Retrieve more rows of a particular query without getting into unnecessary detail about pagination schemes!\n",
+    "\n",
+    "* Authorization: Access more Web APIs quickly! even the ones that implement authorization!\n",
```
Comment (nick-zrymiak) [inline docs/source/user_guide/connector/introduction.ipynb:29]: even -> Even

File: docs/source/user_guide/connector/introduction.ipynb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    "\n",
+    "This section includes a case study for yelp as an example for the process and other sections to explain Connector's functionalities in detail."
```
Comment (nick-zrymiak) [inline docs/source/user_guide/connector/introduction.ipynb:31]: yelp -> Yelp

**Review by nick-zrymiak — COMMENTED** [review]
Format is spot-on!

File: docs/source/user_guide/connector/introduction.ipynb (hunk 2 above)
Comment (peiwangdb) [reply docs/source/user_guide/connector/introduction.ipynb:31]: changed to dblp

**Comment by peiwangdb** [comment]
Thanks for the check @nick-zrymiak! I made changes accordingly and switched the case study from Yelp to dblp.

**Review by dovahcrow — APPROVED** [review]

**Pull request merged**